dbt-bigquery/src/dbt/adapters/bigquery/python_submissions.py

```diff
@@ -163,3 +179,141 @@ def _update_batch_from_config(
+        blob_name = "/".join(blob_names)
+
+        bucket = self._storage_client.bucket(bucket_name)
+        blob = bucket.blob(blob_name)
+
+        try:
+            file_content = blob.download_as_text()
+            data = json.loads(file_content)
+
+        except json.JSONDecodeError:
+            _logger.debug(f"Error: File '{gcs_uri}' content is not valid JSON.")
+            data = None
+
+        except Exception as e:
+            _logger.exception(f"Error reading file from GCS: {e}")
+            data = None
+
+        return data
+
+    def _submit_bigframes_job(self, notebook_template_id: str) -> None:
+
+        notebook_execution_job = aiplatform_v1.NotebookExecutionJob()
+        notebook_execution_job.notebook_runtime_template_resource_name = (
+            f"projects/{self._project}/locations/{self._region}/"
+            f"notebookRuntimeTemplates/{notebook_template_id}"
+        )
+
+        notebook_execution_job.gcs_notebook_source = (
+            aiplatform_v1.NotebookExecutionJob.GcsNotebookSource(uri=self._gcs_location)
+        )
+
+        # TODO(jialuo): Add a method for it.
+        if hasattr(self._GoogleCredentials, "_service_account_email"):
+            notebook_execution_job.service_account = self._GoogleCredentials._service_account_email
+        else:
+            from google.auth.transport.requests import Request
+
+            request = Request()
+            response = request(
+                method="GET",
+                url="https://www.googleapis.com/oauth2/v2/userinfo",
+                headers={"Authorization": f"Bearer {self._GoogleCredentials.token}"},
+            )
+            notebook_execution_job.execution_user = json.loads(response.data).get("email")
+
+        notebook_execution_job.gcs_output_uri = (
+            f"gs://{self._gcs_bucket}/{self._model_file_name}/logs"
+        )
+        notebook_execution_job.display_name = self._get_batch_id()
+        request = aiplatform_v1.CreateNotebookExecutionJobRequest(
+            parent=f"projects/{self._project}/locations/{self._region}",
+            notebook_execution_job=notebook_execution_job,
+        )
+
+        res = self._ai_platform_client.create_notebook_execution_job(request=request).result()
+
+        job_id = res.name.split("/")[-1]
+        gcs_log_uri = f"{notebook_execution_job.gcs_output_uri}/{job_id}/{self._model_name}.py"
+        gcs_log = self._read_json_from_gcs(gcs_log_uri)
+        if gcs_log:
+            # TODO(jialuo): improve the logger info.
+            # TODO(jialuo): raise errors here. There are some situations when
+            # the notebook failed but the pipeline still showed success.
+            _logger.info(
+                f"The colab notebook runtime outputs from GCS: {gcs_log['cells'][0]['outputs']}"
+            )
+        else:
+            _logger.debug(f"Failed to read log from GCS URI: {gcs_log_uri}")
+
+        _ = self._ai_platform_client.get_notebook_execution_job(name=res.name)
```

Comment:
Does this specifically wait for the job to complete? Or is it more of a status check? Our current approach is to wait for the job to complete so that we know we can run downstream models.

Also, can we return the job here? In the two dataproc submission methods we return the Job and the Batch, respectively, during `submit`. I'm not sure how this gets used off the top of my head, but it would be consistent and could make testing and troubleshooting a little easier.